### ISSUE TITLE
why useEventCallback's inputs$ and states$ opposite to their input argument order

### ISSUE THREAD

**Issue by maple-leaf** [LeetCode-OpenSource/rxjs-hooks#53]
```javascript
  const [onEvent] = useEventCallback((events$, x$, y$) => {
    x$.subscribe(x => console.log('x ', x)) // log: x [false]
    y$.subscribe(y => console.log('y ', y)) // log: y [0]
  }, [0], [false]);
```

I thinks x to be `0` and y to be `false` is expected, but `useEventCallback` give opposite result. Am I doing something wrong.

versions info:
rxjs-hooks: 0.4.3

**Comment by Luefeng** [comment]
in the  source code，the two params's  order is reversed.
```javascript
export type EventCallback<EventValue, State, Inputs> = Not<
  Inputs extends void ? true : false,
  (
    eventSource$: Observable<EventValue>,
    inputs$: Observable<RestrictArray<Inputs>>,
    state$: Observable<State>,
  ) => Observable<State>,
  (eventSource$: Observable<EventValue>, state$: Observable<State>) => Observable<State>
>
```

```javascript
if (!inputs) {
      value$ = (callback as EventCallback<EventValue, State, void>)(event$, state$ as Observable<State>)
    } else {
      value$ = (callback as any)(event$, inputs$ as Observable<Inputs>, state$ as Observable<State>)
    }
```

what's the purpose?:thinking:

**Comment by crimx** [comment]
I am a bit confused.

https://github.com/LeetCode-OpenSource/rxjs-hooks/blame/master/README.md#L338

https://github.com/LeetCode-OpenSource/rxjs-hooks/blame/master/README.md#L217

These two signatures are not the same right?

**Comment by crimx** [comment]
Oh I see. It is reversed in `EventCallback`.

https://github.com/LeetCode-OpenSource/rxjs-hooks/blame/master/README.md#L203-L207

**Comment by crimx** [comment]
But if I write something like this the `EventValue` generic of `useEventCallback` gets resolved to `unknown`. How to fix it?

```js
const [onChange] = useEventCallback(
  (event$: Observable<React.MouseEvent<HTMLDivElement>>, $input) =>
    event$.pipe(
      map(e => e.type === 'mouseenter'),
      debounceTime(300),
      withLatestFrom($input),
      tap(([type , [onChange]]) => onChange(type))
    ),
  false,
  [props.onChange]
)
```

**Comment by OBe95** [comment]
@Brooooooklyn I would like to PR this, however it would be a breaking change as the order should be reversed for each call that uses the 3 observables

**Comment by zry656565** [comment]
Fixed in #115
